exercicios/projeto-casa/index.js

```diff
@@ -0,0 +1,141 @@
+/*- Criar os clientes do banco
+- Atualizar informações desses clientes ( como endereço, telefone de contato...)
+- Fazer depósitos / pagamentos usando o saldo de sua conta
+- Encerrar contas de clientes
+- Conseguir Filtrar os clientes do banco pelo seu nome,por saldo...*/
+
+
+
+
+const express = require ("express")
+const app = express()
+const port = 3002;
+const uuid = require ("uuid")
+const IDUnico = uuid.v4()
+const GerarNovoNumeroConta = Math.random();
+const listaClientesBanco = require("./model/contas-clientes")
+
+app.use(express.json());
+
+
+const ordenandoContasClientes = () => {
+    const novaListaOrdenadaDeContas = listaClientesBanco.sort((clienteA, clienteB) => {
+      if (clienteA.data_nascimento < clienteB.data_nascimento) {
+        return -1;
+      }
+      if (clienteA.data_nascimento > clienteB.data_nascimento) {
+        return 1;
+      }
+  
+      return 0;
+    });
+  
+    const valorUltimoID = novaListaOrdenadaDeContas[novaListaOrdenadaDeContas.length - 1];
+  
+    return valorUltimoID.index + 1;
+    
+  };
+  
+  console.log()
+
+//Criar os clientes do banco, foi utilizado o método post, que (concluido)
+
+
+app.post("/clientes/add",(req,res) => {
+    const {id,nome_cliente, cpf_cliente, cliente,data_nascimento,
+        conta:{numero,tipo,saldo,data_criação}} = req.body;
+  
+    const dataAberturaConta = new Date();
+    const clienteNovo = ordenandoContasClientes();
+    const novoCliente = {
+        
+        id : uuid.v4(),
+        nome_cliente  ,
+        cpf_cliente,
+        cliente,
+        data_nascimento,
+conta:{
+        numero:Math.random(),
+        tipo ,
+        saldo,
+        data_criação :dataAberturaConta.toISOString(),
+    
+},
+    }
+    listaClientesBanco.push(ordenandoContasClientes)
+return res.json(novoCliente)
+    
+});
+
+
+//- Atualizar informações desses clientes ( telefone de contato e tipo de conta atualizados...(Concluido))
+
+     
+app.patch("/cliente/:id",(req,res) => {
+
+    const idUsuario = req.params.id
+    const dadoAlterado = req.body

```

Comment:
Mapeie os registros que virá daqui através do destruct do objeto.

1. Porque você não quer que ele envie , por exemplo, o valor do saldo dele na conta
2. porque dessa forma você não quebra seu objeto quando for fazer o spread